docs/core/extensions/dependency-injection-auto-activation.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Dependency injection auto-activation
+description: Learn how to use auto-activation for singleton services in .NET to ensure services are instantiated at startup rather than on first use.
+author: IEvangelist
+ms.author: dapine
+ms.date: 10/20/2025
+ms.topic: concept-article
+ai-usage: ai-assisted
+---
+
+# Dependency injection auto-activation
+
+The [`Microsoft.Extensions.DependencyInjection.AutoActivation`](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection.AutoActivation) NuGet package provides functionality to automatically activate singleton services at application startup, rather than waiting for their first use. This approach can help minimize latency for initial requests by ensuring services are ready to handle requests immediately when the application starts.
+
+## When to use auto-activation
+
+Auto-activation is beneficial in scenarios where:
+
+- You want to minimize the latency of the first request by avoiding the overhead of lazy initialization.
+- Services need to perform initialization work at startup, such as warming up caches or establishing connections.
+- You want consistent response times from the start of your application's lifetime.
+- Services have expensive constructors that you want to execute during startup rather than during request processing.
+
+## Get started
+
+To get started with auto-activation, install the [`Microsoft.Extensions.DependencyInjection.AutoActivation`](https://www.nuget.org/packages/Microsoft.Extensions.DependencyInjection.AutoActivation) NuGet package.
+
+### [.NET CLI](#tab/dotnet-cli)
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.DependencyInjection.AutoActivation
+```
+
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="Microsoft.Extensions.DependencyInjection.AutoActivation" Version="9.8.0" />
+```
+
+---
+
+For more information, see [dotnet add package](../tools/dotnet-add-package.md) or [Manage package dependencies in .NET applications](../tools/dependencies.md).
+
+## Register services for auto-activation
+
+The auto-activation package provides extension methods to register services that are automatically activated when the service provider is built. There are several ways to register services for auto-activation:
+
+### AddActivatedSingleton
+
+The <xref:Microsoft.Extensions.DependencyInjection.AutoActivationExtensions.AddActivatedSingleton%2A> method registers a service as a singleton and ensures it's activated at startup:
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+var services = new ServiceCollection();
+
+services.AddActivatedSingleton<MyService>();
+
+var provider = services.BuildServiceProvider();
+
+// MyService is already instantiated at this point
```

Comment:
No, this won't instantiate MyService.

AutoActivation works by registering an IHostedService whose StartAsync method requests all the auto-activated services from the IServiceProvider.  However, this sample does not start any IHostedService.